[Tooling] Add Localization sync job

iangmaia opened the pull request:
Related to [AINFRA-573: Integrate Fluent translations import/export in the Release Process](https://linear.app/a8c/issue/AINFRA-573/integrate-fluent-translations-importexport-in-the-release-process)

Based on https://github.com/Automattic/wordpress-rs/pull/765, which needs to be merged first.

This PR adds a job to sync the translations, **reusing the nightly schedule**. It essentially:
1. Generates a new source English `.po` file based on `wp_localization/localization/en-US/main.ftl` and updates it in the repo (so it is uploaded to GlotPress with a wpcom job)
2. Downloads the translations from GlotPress and converts them to Fluent files, updating them in the repo.

iangmaia committed:
Use latest release-toolkit Gem

iangmaia committed:
Add Fluent / PO localization lanes

iangmaia committed:
Update source PO file (en-US.pot) to be synced to GlotPress

iangmaia committed:
Update generated Fluent files based on latest GlotPress translations

iangmaia committed:
Update release-toolkit

iangmaia committed:
Fix Rubocop violations

iangmaia committed:
Run bundle update

iangmaia committed:
Create temporary dir for downloading PO files and use open-uri for downloading

iangmaia committed:
Update code to use `fluent-tools` Gem

iangmaia committed:
Update generated source PO file

iangmaia committed:
Update generated Fluent files based on latest GlotPress translations

iangmaia committed:
Add job and script to sync localization files

AliSoftware commented on .buildkite/commands/sync-localization.sh:
```diff
@@ -0,0 +1,13 @@
+echo "--- :rubygems: Setting up Gems"
+install_gems
+
+echo "--- :globe_with_meridians: :arrow_up: Generate the source language PO file for GlotPress based on `wp_localization/localization/en-US/main.ftl`"
+bundle exec fastlane generate_source_po_file commit_and_push_changes:true
+
+echo "--- :globe_with_meridians: :arrow_down: Download and update translations from GlotPress and update the local Fluent files"
+bundle exec fastlane download_translations commit_and_push_changes:true
```
Not critical, but maybe it'd be nicer to have the lanes only `commit` without pushing, and push only at the end after both lanes have run, to push both commits at once instead of having 2 push operations? 🤷

iangmaia replied:
Yeah, I also realized it once I published this PR 👍  I wasn't sure how the lanes were going to work independently but having them running together makes it clearer that is better to avoid the side effect in the lane itself.

I'll write it down to tackle it in https://github.com/Automattic/wordpress-rs/pull/765 and then update it here 👍

iangmaia replied:
Updated in 2c4547caa946d6d19c5e2a89dc7cd6e1d7fcaac2 / e022478ba7688903d662737b01c8bd692c835c6c.

iangmaia committed:
Update lanes to not push automatically

iangmaia committed:
Update sync script to only push at the end

iangmaia committed:
Add localization README file

iangmaia committed:
Reorder helper lane

iangmaia committed:
Validate file_path at the beginning of the `generate_fluent_file_from_po` lane

iangmaia committed:
Update code to exit early in `download_translations` when there are no downloaded files

iangmaia committed:
Add all locales supported by the iOS and Android apps
h/t to @crazytonyli for highlighting that the supported locales should be a combination of the locales supported in the iOS and Android apps https://github.com/Automattic/wordpress-rs/pull/765#discussion_r2151887208

iangmaia committed:
Add generated Fluent files with the updated list of supported locales

iangmaia committed:
Update fluent-tools to 0.2.0 and re-generate localization files with the latest version

iangmaia committed:
Update to 0.3.0 release

iangmaia committed:
Update PO file with latest fluent-tools

iangmaia committed:
Merge branch 'trunk' into iangmaia/fluent-to-glotpress-with-rust-tools

iangmaia committed:
Merge branch 'iangmaia/fluent-to-glotpress-with-rust-tools' into iangmaia/localization-sync-job

iangmaia committed:
Add IMAGE to nightly localization sync job

iangmaia committed:
Add ci-toolkit plugin to localization sync job

iangmaia committed:
Make sure to checkout branch (given we're pushing later)

wpmobilebot committed:
Update source PO file (en-US.pot) to be synced to GlotPress

iangmaia committed:
Merge branch 'trunk' into iangmaia/localization-sync-job

iangmaia committed:
Merge branch 'trunk' into iangmaia/localization-sync-job
